R/raincloudPlots.R

```diff
@@ -0,0 +1,908 @@
+  vioSides      <- getVioSides$sides
+  errorVioSides <- getVioSides$error
+
+  boxPosVec   <- .rainNudgeForEachCloud(options$boxNudge, vioSides)
+  boxPosition <- ggpp::position_dodge2nudge(  # boxPosition for whiskers and .rainGeomRain
+    x        = boxPosVec,
+    width    = 0,
+    padding  = options$boxPadding,
+    preserve = "single"  # All boxes same width and different amounts of boxes are centered around middle
+  )
+
+  # .rainGeomRain() - workhorse function, uses ggrain::geom_rain()
+  plotInProgress <- plotInProgress + .rainGeomRain(
+    dataset, options, infoFactorCombinations, vioSides, boxPosition, plotInProgress
+  )
+
+  # Whiskers for boxplots
+  boxDataIndex   <- if (options$observationId == "") 2 else 3
+  boxData        <- ggplot2::ggplot_build(plotInProgress)$data[[boxDataIndex]]
+  getWhiskers    <- .rainWhiskers(options, boxData, infoFactorCombinations, boxPosition)
+  plotInProgress <- plotInProgress + getWhiskers$lowerWhiskers + getWhiskers$upperWhiskers
+
+  # Means and Lines
+  meanPosition <- if (options$meanPosition == "likeBox") {
+    ggpp::position_dodge2nudge(x = boxPosVec, width = options$boxWidth, preserve = "single")
+  } else {
+    ggpp::position_dodge2nudge(x = 0, width = 0.00000000000001, preserve = "single")
+    # With the default "identity" as position, colors of meanLines would be scrambled
+    # and width = 0 messes up the position of the means (same for several position_ functions I tried)
+    # Thus, set very small width that the human eye will not notice.
+  }
+  getMeansAndLines <- .rainMeansAndLines(options, boxPosVec, aesX, aesFill, infoFactorCombinations, meanPosition)
+  plotInProgress <- plotInProgress + getMeansAndLines$meanLines + getMeansAndLines$means  # Lines first so means cover
+
+  # Interval around mean
+  intervalPosition <- if (options$meanPosition == "likeBox") meanPosition else "identity"
+  if (options$meanInterval || options$meanIntervalCustom) {
+    intervalBounds <- dataInfo$intervalBounds[[inputVariable]]
+      if (intervalBounds$successfulComputation) {
+        lowerBound <- intervalBounds$lowerBound
+        upperBound <- intervalBounds$upperBound
+        meanInterval <- ggplot2::stat_summary(
+          ggplot2::aes(ymin = ..y.. - (..y.. - lowerBound), ymax = ..y.. + (upperBound - ..y..)),
+          fun         = mean,
+          geom        = "errorbar",
+          width       = options$boxWidth,
+          lwd         = options$boxOutlineWidth,
+          position    = intervalPosition,
+          color       = .rainOutlineColor(options, "colorPalette", infoFactorCombinations),
+          show.legend = FALSE
+        )
+      } else {
+        meanInterval <- NULL
+      }
+  } else {
+    meanInterval <- NULL
+  }
+  plotInProgress <- plotInProgress + meanInterval
+
+
+  # Horizontal plot?
+  if (options$horizontal) plotInProgress <- plotInProgress + ggplot2::coord_flip()
```

Comment:
place on separate line

Reply:
The JASP r-style-guide allows single-line if-else statements:
https://github.com/jasp-stats/jasp-desktop/blob/development/Docs/development/r-style-guide.md
![Screenshot 2024-04-15 at 16 41 17](https://github.com/jasp-stats/jaspDescriptives/assets/123652963/3b3f6be5-26c7-4dbb-a65e-11cdcd634105)

Reply:
ok, good (I tagged this as it looked quite long)